python/cog/server/worker.py

```diff
@@ -458,22 +472,13 @@ def _load_predictor(self) -> Optional[BasePredictor]:
 
         return None
 
-    def _setup(
-        self, redirector: Union[StreamRedirector, SimpleStreamRedirector]
-    ) -> None:
-        done = Done()
-        try:
+    def _validate_predictor(
+        self,
+        redirector: Union[StreamRedirector, SimpleStreamRedirector],
+    ) -> bool:
+        with self._handle_setup_error(redirector):
             assert self._predictor
 
-            # Could be a function or a class
-            if hasattr(self._predictor, "setup"):
-                if not has_setup_weights(predictor):
-                    predictor.setup()
-                    return
-
-                weights = extract_setup_weights(predictor)
-                predictor.setup(weights=weights)  # type: ignore
-
             predict = get_predict(self._predictor)
 
             is_async_predictor = inspect.iscoroutinefunction(
```

Comment:
this exactly duplicates `self._is_async`. I think we should get rid of this local variable and rename `self._is_async` to `self._is_async_predictor`. (I know your code didn't introduce this but it confused me while reviewing this PR)